kiwmi/input/cursor.c

```diff
@@ -284,65 +284,15 @@ cursor_refresh_focus(
     struct kiwmi_desktop *desktop = &cursor->server->desktop;
     struct wlr_seat *seat         = cursor->server->input.seat->seat;
 
-    double ox                     = cursor->cursor->x;
-    double oy                     = cursor->cursor->y;
-    struct wlr_output *wlr_output = wlr_output_layout_output_at(
-        desktop->output_layout, cursor->cursor->x, cursor->cursor->y);
-
-    wlr_output_layout_output_coords(
-        desktop->output_layout, wlr_output, &ox, &oy);
-
-    struct kiwmi_output *output = wlr_output->data;
-
     struct wlr_surface *surface = NULL;
     double sx;
     double sy;
 
-    struct kiwmi_layer *layer = layer_at(
-        &output->layers[ZWLR_LAYER_SHELL_V1_LAYER_OVERLAY],
-        &surface,
-        ox,
-        oy,
-        &sx,
-        &sy);
-    struct kiwmi_view *view;
-
-    if (!layer) {
-        layer = layer_at(
-            &output->layers[ZWLR_LAYER_SHELL_V1_LAYER_TOP],
-            &surface,
-            ox,
-            oy,
-            &sx,
-            &sy);
-    }
-
-    if (!layer) {
-        view = view_at(
-            desktop, cursor->cursor->x, cursor->cursor->y, &surface, &sx, &sy);
-    }
-
-    if (!layer) {
-        layer = layer_at(
-            &output->layers[ZWLR_LAYER_SHELL_V1_LAYER_BOTTOM],
-            &surface,
-            ox,
-            oy,
-            &sx,
-            &sy);
-    }
-
-    if (!layer) {
-        layer = layer_at(
-            &output->layers[ZWLR_LAYER_SHELL_V1_LAYER_BACKGROUND],
-            &surface,
-            ox,
-            oy,
-            &sx,
-            &sy);
-    }
+    // TODO: erm, shouldn't we ask for any surface instead? /shrug
+    struct kiwmi_toplevel *toplevel = toplevel_at(
```

Comment:
I’m not quite sure about this one. If it’s about focusing a surface, should we really only ask for toplevels? (Although there isn’t any surface that couldn’t get us to a toplevel atm.)